api/scpca_portal/models/library.py

```diff
@@ -58,17 +62,52 @@ def get_from_dict(cls, data):
     def bulk_create_from_dicts(cls, library_jsons: List[Dict], sample) -> None:
         libraries = []
         for library_json in library_jsons:
-            libraries.append(Library.get_from_dict(library_json))
+            if not Library.objects.filter(scpca_id=library_json["scpca_library_id"]).exists():
+                libraries.append(Library.get_from_dict(library_json))
```

Comment:
```suggestion
                # TODO: remove when scpca_project_id is in source json
                library_json["scpca_project_id"] = sample.project.scpca_id
                libraries.append(Library.get_from_dict(library_json))
```
I was thinking we should break up the `get_modality_and_formats` method into two. But then really, at that point we should just be updating the everything from the context of a single Library instance. Let's drop the static methods and place them directly in `get_from_dict`.

I think this is worth putting in there to avoid accidentally creating a Library instance that is in an "incomplete" state.